substrate/frame/preimage/precompiles/src/interface/IPreimage.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.30;
+
+/// @dev The on-chain address of the Preimage precompile.
+address constant PREIMAGE_PRECOMPILE_ADDRESS = address(0xD0000);
+
+/// @title `pallet_preimage` Precompile Interface
+/// @notice An interface for interacting with `pallet_preimage`.
+/// It forwards calls directly to the corresponding dispatchable functions.
+interface IPreimage {
+	/// @notice Register a preimage on-chain.
+	/// @dev This transaction is free when called by a manager or when `preimage` was previously requested.
+	/// @param preimage The preimage to be registered on-chain.
+	/// @custom:reverts If `preimage` is larger than the allowed `CALLDATA_BYTES`.
+	/// @custom:reverts If `preimage` is already noted when called by a non-manager.
+	/// @custom:reverts If caller has unsufficient funds for deposit.
```

Comment:
How would the caller know why it reverts? Do they get the revert string?  
I am just wondering since the case that it is already noted will probably be handled differently than the other cases.

Reply:
Related to the other discussion: Quoted from [this](https://forum.polkadot.network/t/can-pallet-revive-help-improve-polkadots-ux/12923/18?u=lukk) forum comment:
> We need to think hard of how level level the pre-compile API can be so we can guarantee its stability.[...]We should expose as much as we have to and as little as we can to still allow writing a useful frontend. This includes simple things like error codes: In the pre-compile we should just map them to success/failure since it will be hard to keep them stable. Error handling should not happen on-chain.